astroplan/core.py

```diff
@@ -37,70 +54,94 @@ def __init__(self, name, longitude, latitude, elevation, pressure,
 
-        longitude : str or `~astropy.units.quantity`?
-            The longitude of the observing location.
+        pressure : `~astropy.units.Quantity`
+            The ambient pressure.
+
+        location : `~astropy.coordinates.EarthLocation`
+            The location (latitude, longitude, elevation) of the observatory.
+
+        longitude : str or `~astropy.units.quantity`
+            The longitude of the observing location. If str, should be a string
+            that initializes a `~astropy.coordinates.Longitude` object with
+            units in degrees.
 
-        latitude : str or `~astropy.units.quantity`?
-            The latitude of the observing location.
+        latitude : str or `~astropy.units.quantity`
+            The latitude of the observing location. If str, should be a string
+            that initializes a `~astropy.coordinates.Latitude` object with
+            units in degrees.
 
         elevation : `~astropy.units.Quantity`
             The elevation of the observing location, with respect to sea
             level.
 
-        pressure : `~astropy.units.Quantity`
-            The ambient pressure.
-
         relative_humidity : float
             The ambient relative humidity.
 
         temperature : `~astropy.units.Quantity`
             The ambient temperature.
 
-        timezone : WHAT TYPE IS pytz.timezone ?
-            The local timezone.
+        timezone : str or `datetime.tzinfo`
+            The local timezone, as either an instance of `pytz.timezone()` or
+            the string accepted by `pytz.timezone()`.
 
         description : str
             A short description of the telescope, observatory or observing
             location.
         """
-        raise NotImplementedError()
 
-    def set_environment(self, pressure, relative_humidity, temperature):
-        """
-        Updates the Observer object with environmental conditions.
+        if name is not None:
+            self.name = name
 
-        <longer description>
+        # If lat/long given instead of EarthLocation, convert them
+        # to EarthLocation
+        if location is None and (latitude is not None and longitude is not None):
+            accepted_latlon_types = [str, Quantity]
+            if (type(latitude) in accepted_latlon_types and
+                type(longitude) in accepted_latlon_types):
+                latitude = Latitude(latitude, unit=u.degree)
+                longitude = Longitude(longitude, unit=u.degree)
 
-        Parameters
-        ----------
-        pressure : `~astropy.units.Quantity`
-            The ambient pressure.
+            self.location = EarthLocation.from_geodetic(longitude, latitude,
+                                                        elevation)
 
-        relative_humidity : `~astropy.units.Quantity`
-            The ambient relative humidity.
+        elif isinstance(location, EarthLocation):
+            self.location = location
```

Comment:
This is my compromise between @ejeschke 's accepting attitude towards input formats and @adrn 's [preference for accepting one type (`EarthLocation`)]([-style extreme](https://github.com/astroplanners/astroplan/pull/4#discussion_r32672422) ). I accept latitudes and longitudes or an `EarthLocation`. If you input latitude/longitude, they become an EarthLocation under the hood.